apps/testingaccessibility/src/pages/invoices/_middleware.ts

```diff
@@ -0,0 +1,16 @@
+import {NextResponse} from 'next/server'
+import type {NextFetchEvent, NextRequest} from 'next/server'
+import {NextApiRequest} from 'next'
+import {getDecodedToken} from '../../utils/get-decoded-token'
+
+export async function middleware(req: NextApiRequest, ev: NextFetchEvent) {
+  const session = await getDecodedToken(req)
```

Comment:
this approach doesn't require any network traffic and decodes the JWT from cookies if available so we can tell if they are logged in.